util/deephash/deephash.go

```diff
@@ -525,9 +525,13 @@ func (h *hasher) hashComplex128v(v addressableValue) bool {
 // hashString hashes v, of kind time.Time.
 func (h *hasher) hashTimev(v addressableValue) bool {
 	t := *(*time.Time)(v.Addr().UnsafePointer())
-	b := t.AppendFormat(h.scratch[:1], time.RFC3339Nano)
-	b[0] = byte(len(b) - 1) // more than sufficient width; if not, good enough.
-	h.HashBytes(b)
+	h.HashUint64(uint64(t.Unix()))
+	h.HashUint32(uint32(t.Nanosecond()))
+	// TODO(joetsai): Do we care about the time zone? Including it means that
+	// t1.Equal(t2) is not identical to Hash(t1) == Hash(t2),
+	// which is an odd semantic.
+	_, offset := t.Zone()
+	h.HashUint32(uint32(offset))
```

Comment:
Personally, I'm surprised that we include the time zone as part of the hash. I vote that we remove it. However, I don't know whether any usages of `deephash` depend on the zone being part of the comparison or not.

Reply:
Remember that deephash was originally defined as the SHA256 of the JSON we sent to clients, because the protocol is JSON, and if the JSON changed, we wanted the hash to change.

With that in mind, consider:

https://go.dev/play/p/n17DwNjMp_q

Those times are equal, but have different JSON.

Also keep in mind that the Swift date/time parser is very picky and we've had 2 or 3 big bugs in the past because we sent the wrong time.Time down to Go which passed through to Swift and broke the Mac GUI.

So I'd like to continue to be paranoid about hashing everything in a time.Time.

Reply:
Done. I documented the desired semantic for `time.Time`.

Reply:
And added a fuzz test to verify that `Hash(t1) == Hash(t2)` is semantically equivalent to `t1.Format(RFC3339Nano) == t2.Format(RFC3339Nano)`.

I'll run it for an hour before submitting.

Reply:
1 hour later. 650M timestamps fuzzed, no issues.

Reply:
Lgtm